go/pools/resource_pool.go

```diff
@@ -104,9 +104,12 @@ func (rp *ResourcePool) TryGet() (resource Resource, err error) {
 
 func (rp *ResourcePool) get(ctx context.Context, wait bool) (resource Resource, err error) {
 	// If ctx has already expired, avoid racing with rp's resource channel.
-	if ctx.Err() != nil {
+	select {
+	case <-ctx.Done():
```

Comment:
The intention is to test whether ctx is canceled or not. Based on following doc, ctx.Err() will return Canceled if ctx has already expired. In other words, we want to return ErrTimeout once deadline is reached (even if ctx is not Done).

  // Err returns a non-nil error value after Done is closed.  Err returns
    // Canceled if the context was canceled or DeadlineExceeded if the
    // context's deadline passed.  No other values for Err are defined.
    // After Done is closed, successive calls to Err return the same value.
    Err() error